src/Filters/FiltersScope.php

```diff
@@ -3,17 +3,43 @@
 namespace Spatie\QueryBuilder\Filters;
 
 use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
 use Illuminate\Support\Arr;
 use Illuminate\Support\Str;
+use ReflectionObject;
 
 class FiltersScope implements Filter
 {
-    public function __invoke(Builder $query, $values, string $property)
+    public function __invoke(Builder $query, $values, string $property): Builder
     {
         $scope = Str::camel($property);
 
         $values = Arr::wrap($values);
+        $values = $this->resolveTypeHintedParameters($query, $values, $scope);
 
-        $query->$scope(...array_values($values));
+        return $query->$scope(...$values);
+    }
+
+    protected function resolveTypeHintedParameters(Builder $query, $values, string $scope): array
+    {
+        $scopeParameters = (new ReflectionObject($query->getModel()))
+            ->getMethod('scope' . ucfirst($scope))
+            ->getParameters();
+
+        foreach ($scopeParameters as $scopeParameter) {
+            if (! optional($scopeParameter->getClass())->isSubclassOf(Model::class)) {
+                continue;
+            }
+
+            $model = $scopeParameter->getClass()->newInstance();
+            $index = $scopeParameter->getPosition() - 1;
+            $value = $values[$index];
+
+            if (is_numeric($value)) {
+                $values[$index] = $model::findOrFail($value);
```

Comment:
`findOrFail` will return a 404 response. This is not ideal for a query builder endpoint. I'm not sure what the proper response would be.

@michapietsch mentioned an empty result set but that might be confusing as to why it's empty. Maybe we can just throw a custom exception that explains the error?

Reply:
@AlexVanderbist I agree with custom exception. Maybe with 422 status code?

Reply:
Sounds like a plan!